### PR TITLE
staging deployment [INFORM-2928]

### PR DESCRIPTION
**Pull request by saifmahamood**
## What
We are adding a deployment for staging here via Kustomize files. 

The Kustomize deployment files already exist in the Airbyte repository but we copied them along with the resources into a separate Kustomize repository to create deployments. In addition, an `ingress.yaml` file was added. And the deployment for this is done via a Terraform Kustomize provider under the `helm` directory.

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+          role-skip-session-tagging: true
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: retrieve ecr password and store as output
+        id: ecr_login_pwd
+        run: |
+          PWD=$(aws ecr get-login-password --region us-east-1)
```
Comment (devshah1) [inline .github/workflows/pull_request.yml]: This needs to be us-east-2.

File: deployments/helm/stg/us_east_2/main.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
+locals {
+  k8s_namespace_name_airbyte = "airbyte"
+}
+
+data "aws_eks_cluster" "cluster" {
+  provider = aws
```
Comment (mhemken-vts) [inline deployments/helm/stg/us_east_2/main.tf]: This one needs to be specified too

File: deployments/helm/stg/us_east_2/versions.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+terraform {
+  required_version = "~> 0.15.1"
+  required_providers {
+    aws = {
+      version = "~> 3.46.0"
+    }
+
+    kustomization = {
+      source  = "kbst/kustomize"
+      version = "0.2.0-beta.3"
```
Comment (mhemken-vts) [inline deployments/helm/stg/us_east_2/versions.tf]: version = "0.7.0"

Comment (saifmahamood) [reply deployments/helm/stg/us_east_2/versions.tf]: `│ Could not retrieve the list of available versions for provider kbst/kustomize: no available releases match the given constraints 0.7.0`

**Pull request merged**